### PR TITLE
CXP-1422: Credentials regeneration

### PR DESCRIPTION
**Pull request by eoliveauakeneo**
**Description (for Contributor and Core Developer)**

image
image
image

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/CustomApps/Command/RegenerateCustomAppSecretHandler.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Application\CustomApps\Command;
+
+use Akeneo\Connectivity\Connection\Application\RandomCodeGeneratorInterface;
+use Akeneo\Connectivity\Connection\Domain\CustomApps\Persistence\UpdateCustomAppSecretQueryInterface;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Application/CustomApps/Command/RegenerateCustomAppSecretHandler.php:11]:
```suggestion
```

File: src/Akeneo/Connectivity/Connection/back/Application/CustomApps/Command/RegenerateCustomAppSecretHandler.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
```
Comment (guillaumefer-akeneo) [inline src/Akeneo/Connectivity/Connection/back/Application/CustomApps/Command/RegenerateCustomAppSecretHandler.php:12]:
```suggestion
 * @copyright 2023 Akeneo SAS (http://www.akeneo.com)
```

File: src/Akeneo/Connectivity/Connection/back/Domain/CustomApps/Persistence/UpdateCustomAppSecretQueryInterface.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Domain\CustomApps\Persistence;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/Domain/CustomApps/Persistence/UpdateCustomAppSecretQueryInterface.php]: :thinking:

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/CustomApps/Persistence/UpdateCustomAppSecretQueryIntegration.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+        $this->updateCustomAppSecretQuery->execute(
+            clientId: 'wrong_id',
+            clientSecret: 'NjNjZWMxODc2Mjg2ZTkxMjEyYmI3NDMwNzkxZjRkNjQ3MTA5MDdjNWQzMWMzOTU3MTE1YWYxYWVjNDY0MWUwOQ'
+        );
+
+        $this->assertNotSame(
+            $this->getCustomAppSecret('100eedac-ff5c-497b-899d-e2d64b6c59f9'),
+            'NjNjZWMxODc2Mjg2ZTkxMjEyYmI3NDMwNzkxZjRkNjQ3MTA5MDdjNWQzMWMzOTU3MTE1YWYxYWVjNDY0MWUwOQ'
+        );
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/CustomApps/Persistence/UpdateCustomAppSecretQueryIntegration.php:67]:
```suggestion
       $preQuerySecret = $this->getCustomAppSecret('100eedac-ff5c-497b-899d-e2d64b6c59f9');
        $this->updateCustomAppSecretQuery->execute(
            clientId: 'wrong_id',
            clientSecret: 'NjNjZWMxODc2Mjg2ZTkxMjEyYmI3NDMwNzkxZjRkNjQ3MTA5MDdjNWQzMWMzOTU3MTE1YWYxYWVjNDY0MWUwOQ'
        );
        
        $postQuerySecret = $this->getCustomAppSecret('100eedac-ff5c-497b-899d-e2d64b6c59f9');

        $this->assertSame($preQuerySecret, $postQuerySecret );
```
WDYT?

File: src/Akeneo/Connectivity/Connection/front/src/connect/components/ConnectedApp/Settings/RegenerateSecret.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import {NotificationLevel, useNotify} from '../../../../shared/notify';
+import {useRouter} from '../../../../shared/router/use-router';
+import {useCustomAppRegenerateSecret} from '../../../hooks/use-custom-app-regenerate-secret';
+import {useConnectedApp} from '../../../hooks/use-connected-app';
+import {RegenerateSecretConfirm} from './RegenerateSecretConfirm';
+import {RegenerateSecretNewCredentials} from './RegenerateSecretNewCredentials';
+
+type Step = 'confirm' | 'new_credentials';
+
+export const RegenerateSecret = () => {
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/front/src/connect/components/ConnectedApp/Settings/RegenerateSecret.tsx:14]: Should be a page component under the page directory to remain consistent with other routes

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/components/ConnectedApp/RegenerateSecret.test.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+test('The regenerate secret modal return null if no connectedApp found or still loading', () => {
+    const {container} = renderWithProviders(<RegenerateSecret />);
+    expect(container).toBeEmptyDOMElement();
+});
+
+test('The regenerate secret modal renders the regenerate step by default', async () => {
+    renderWithProviders(
+        <MemoryRouter initialEntries={['connect/connected-apps/custom_app/regenerate-secret']}>
+            <Route path='connect/connected-apps/:connectionCode/regenerate-secret'>
+                <RegenerateSecret />
```
Comment (RiffVentura) [inline src/Akeneo/Connectivity/Connection/front/tests/src/connect/components/ConnectedApp/RegenerateSecret.test.tsx:50]: Why do we need a router here? 
Directly rendering the component should be enough IMO

**Review by RiffVentura — APPROVED** [review]
GTM after fixes

**Review by guillaumefer-akeneo — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/CustomApps/Persistence/UpdateCustomAppSecretQueryIntegration.php (hunk 4 above)
Comment (eoliveauakeneo) [reply src/Akeneo/Connectivity/Connection/back/tests/Integration/CustomApps/Persistence/UpdateCustomAppSecretQueryIntegration.php:67]: That's clearer yep, i'll to that

File: src/Akeneo/Connectivity/Connection/front/tests/src/connect/components/ConnectedApp/RegenerateSecret.test.tsx (hunk 6 above)
Comment (eoliveauakeneo) [reply src/Akeneo/Connectivity/Connection/front/tests/src/connect/components/ConnectedApp/RegenerateSecret.test.tsx:50]: I did not find a more straightforward solution to be able to get my connectionCode with useParams()

**Pull request merged**